server/controllers/assessmentDayDataController/index.js

```diff
@@ -0,0 +1,61 @@
+import AssessmentDayDataModel from '../../models/AssessmentDayDataModel'
+import ToCModel from '../../models/ToC'
+import AssessDayDataService from '../../services/AssessmentDayDataService'
+import _ from 'lodash'
+
+const AssessmentDayDataController = {
+  create: async (req, res) => {
+    try {
+      const { appDb } = req.app.locals
+
+      const assessmentDataService = new AssessDayDataService()
+      const { metadata, subject_assessments } = req.body
+
+      if (metadata && subject_assessments.length) {
+        const { assessment, subject, collection } = metadata
+        const query = {
+          assessment,
+          subject,
+        }
+        const storedAssessmentData = await AssessmentDayDataModel.all(
+          appDb,
+          collection
+        )
+        const newAssessmentData = assessmentDataService.filterNewData(
+          subject_assessments,
+          storedAssessmentData
+        )
+        const updatedData = assessmentDataService.filterUpdatedData(
+          subject_assessments,
+          storedAssessmentData
+        )
```

Comment:
what do you think about passing these params to the service when initializing it? Something like:

```js
const assessmentDataService = new AssessDayDataService(subject_assessments, storedAssessmentData)
const newAssessmentData = assessmentDataService.filterNewData
const updatedData = assessmentDataService.filterUpdatedData
```

Reply:
[c1cf0d0](https://github.com/AMP-SCZ/dpdash/pull/382/commits/c1cf0d04b9d9dd43b6aceef72cc4f1f45af67a9c)